cmd/commands/cmd_payments.go

```diff
@@ -603,6 +648,13 @@ func SendPaymentRequest(ctx *cli.Context, req *routerrpc.SendPaymentRequest,
 		}
 	} else {
 		var err error
+		err = maybeValidateMPPParams(
+			lnwire.MilliSatoshi(req.Amt*1000), req.MaxParts, lnwire.MilliSatoshi(req.MaxShardSizeMsat),
+		)
+		if err != nil {
+			return status.Errorf(codes.InvalidArgument,
+				"invalid MPP parameters: %v", err)
+		}
```

Comment:
note that this only applies validation if used via the CLI. I think we want the validation to apply regardless of how SendPayment is being called. I think a better place for this validation is thus here:

https://github.com/lightningnetwork/lnd/blob/684041bbfe6ac2e28b34b98f64ae718a2b34b807/lnrpc/routerrpc/router_backend.go#L851-L867